cmd/kube-scheduler/app/config/config.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	apiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/informers"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/cmd/controller-manager/app"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+// Config has all the context to run a Scheduler
+type Config struct {
+	// config is the scheduler server's configuration object.
+	ComponentConfig componentconfig.KubeSchedulerConfiguration
```

Comment:
@aveshagarwal @kubernetes/sig-scheduling-feature-requests we need to figure out what is local to the machine the scheduler is running on and what is shared configuration and placed in etcd.  It's not obvious to me where that line is drawn and some things seem unusual.  For instance, there is a file reference for a kubeconfigfile that refers to something on disk.  1. the kubeconfig is needed in order to contact the apiserver to find the configmap config, and 2. I don't think I would expect the disk layout across all schedulers to be consistent.

Reply:
Continuing that thought, I wouldn't expect the bind addresses to be shared either.

I think I would that I want configuration options controlling behavior to be shared amongst all scheduler instances and I want configuration options controlling exposure and connections to be specific to a particular scheduler instance.

This pull has the misfortune of exposing these problems.

Reply:
@deads2k so far there are 2 things the scheduler is referring on disk: 1)  https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/types.go#L47 which is one of options among on-disk file, configmap and algorithmprovider 2) path to kubeconfig:  https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/types.go#L26 

> I don't think I would expect the disk layout across all schedulers to be consistent.

makes sense.

> I think I would that I want configuration options controlling behavior to be shared amongst all scheduler instances and I want configuration options controlling exposure and connections to be specific to a particular scheduler instance.

So that means these options kubeconfigfile path, bindaddresses (health, metrics), QPS, Burst  per instance, and i think all other options can be shared.